temporalio/worker/activity.py

```diff
@@ -82,14 +82,16 @@ def __init__(
 
         # Validate and build activity dict
         self._activities: Dict[str, _ActivityDefinition] = {}
-        for name, activity in activities.items():
-            if not callable(activity):
-                raise TypeError(f"Activity {name} is not callable")
-            elif not activity.__code__:
-                raise TypeError(f"Activity {name} does not have __code__")
-            elif activity.__code__.co_kwonlyargcount:
-                raise TypeError(f"Activity {name} cannot have keyword-only arguments")
-            elif not inspect.iscoroutinefunction(activity):
+        for activity in activities:
+            # Confirm name is present
+            name: Optional[str] = getattr(activity, "__temporal_activity_name", None)
+            if not name:
+                raise TypeError(
+                    "Activity missing attributes, was it decorated with @activity.defn?"
```

Comment:
I'd say which activity for better dx

Reply:
I wanted to, but I don't know which activity. Are you saying something like "activity #N" where `N` is `index + 1`?

Reply:
You can just print `activity` if it's a function it'll include the name

Reply:
A local function gives something gross like `<function test_activity_without_decorator.<locals>.say_hello at 0x0000020D5FC33D90>` for that. But I am extracting the name now.

Reply:
Done